src/Network/WebSockets/TSSocketManager.m

```diff
@@ -87,6 +88,21 @@ + (void)becomeActive
 
 - (void)becomeActive
 {
+    OWSAssert([NSThread isMainThread]);
```

Comment:
I've moved all of the logic in this class that use class state to the main thread. Most everything was already there, so I don't think I've moved anything expensive.